### PR TITLE
fix jet_puck

### PR DESCRIPTION
**Pull request by eamsden**

### CODE REVIEW

**Review by matthew-levan — APPROVED** [review]

**Review by ashelkovnykov — CHANGES_REQUESTED** [review]

File: rust/ares/src/jets/lock/ed.rs [hunk 1]
```diff
@@ -135,17 +139,15 @@ mod tests {
     fn test_puck() {
```
[the remark this answers is not in the document]
Comment (matthew-levan) [reply rust/ares/src/jets/lock/ed.rs:139]: `tests/sys/zuse/ed25519.hoon`

File: rust/ares/src/interpreter.rs [hunk 2]
```diff
@@ -1225,6 +1225,38 @@ pub fn inc(stack: &mut NockStack, atom: Atom) -> Atom {
...
+        }
+        Right(cell) => {
+            if depth > 0 {
+                assert_all_normalized(cell.head(), path, depth - 1);
+                assert_all_normalized(cell.tail(), path, depth - 1);
+            }
+        }
+    }
+}
+
```
[the remark this answers is not in the document]
Comment (eamsden) [reply rust/ares/src/interpreter.rs:1239]: yes it should

File: rust/ares/src/jets/warm.rs [hunk 3]
```diff
@@ -133,11 +133,16 @@ impl Warm {
...
+        stack: &mut NockStack,
+        s: &mut Noun,
+        f: &mut Noun,
+    ) -> Option<(Jet, Noun)> {
         let warm_it = self.0.lookup(stack, f)?;
-        for (_path, batteries, jet) in warm_it {
+        for (path, batteries, jet) in warm_it {
             if batteries.matches(stack, *s) {
-                return Some(jet);
+                return Some((jet, path));
```
[the remark this answers is not in the document]
Comment (eamsden) [reply rust/ares/src/jets/warm.rs:145]: it is leftover from debugging, but should stay. It's helpful when debugging jet misses.

File: rust/ares/src/jets/lock/ed.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (ashelkovnykov) [reply rust/ares/src/jets/lock/ed.rs:139]: That's not actually the case: neither the old test value nor the new test value are from that test file (which uses the test vectors from [RFC 8032](https://tools.ietf.org/html/rfc8032#section-7.1)).

I double-checked the test values using `puck:ed:crypto` in a fake `zod` with Serf and left a "TODO" for expanding the test suite using the aforementioned vectors.

**Review by ashelkovnykov — APPROVED** [review]

**Pull request merged**